naima/models.py

```diff
@@ -431,3 +431,60 @@ def __call__(self, e):
         e = _validate_ene(e)
         interpy = np.power(10, self._interplogy(np.log10(e.to('eV').value)))
         return self.amplitude * interpy * self.unit
+      
+      
+class EblAbsorptionModel(TableModel):
+    """
+    A TableModel containing the different absorption values from a specific model.
+
+    It returns dimensionless opacity values, that could be multiplied to any model.
+    
+    Parameters
+    ----------    
+    redshift : float
+        Redshift considered for the absorption evaluation.
+        The minimum value used is z = 0.01; Lower values will
+        be considered as redshift 0.
+        
+    ebl_absorption_model : string 
+        Name of the EBL absorption model to use. It may be:
+    
+    * A string equal to ``Dominguez`` (default), containing Dominguez 2011 EBL model
+    
+    NOTE: Current implementation does NOT perform an interpolation in the redshift, so 
+    it just uses the closes z value from the finely binned tau_dominguez11.out file 
+    (delta_z=0.01)    
+        
+    """
+
+    def __init__(self, redshift, ebl_absorption_model='Dominguez'):
+        from scipy.interpolate import interp1d
+        import os
+  
+        self.redshift = validate_scalar('redshift',
+                                   redshift.value,
```

Comment:
That is actually the first thing I tried, but I was getting:
"TypeError: redshift should be a scalar floating point value"
This was happening probably because I was not using any physical_type. I added the physical_type='dimensionless' to the scalar validator. 

Although now it requires a Quantity, and prompts an error when adding a float as an input... Not sure if that is what we want.